darts/models/forecasting/chronos2_model.py

```diff
@@ -0,0 +1,865 @@
+        x_past, x_future, _ = x_in
+        # According to `self._process_input_batch()` in `PLForecastingModule`,
+        # x_past is a stack of [past_target, past_covariates, historic_future_covariates],
+        # while x_future is just future_covariates.
+        # So here we need to create `future_covariates` in Chronos2's format that is
+        # a stack of [past_target (NaNs), past_covariates (NaNs), future_covariates].
+        batch_size, past_length, n_variables = x_past.shape
+        future_length = self.output_chunk_length or 0
+        future_covariates = torch.full(
+            (batch_size, future_length, n_variables),
+            torch.nan,
+            device=x_past.device,
+        )
+        if x_future is not None:
+            n_future_covs = x_future.shape[-1]
+            future_covariates[:, :, -n_future_covs:] = x_future
+
+        # reshape x_past and future_covariates to (batch * vars, time)
+        context = x_past.permute(0, 2, 1).reshape(-1, past_length)
+        future_covariates = future_covariates.permute(0, 2, 1).reshape(
+            -1, future_length
+        )
+
+        # create group_ids according to sample index within the batch
+        group_ids = torch.arange(batch_size, device=context.device).repeat_interleave(
+            n_variables
+        )
+
+        # determine minimum number of patches to cover future_length
+        num_output_patches = math.ceil(
+            future_length / self.chronos_config.output_patch_size
+        )
+
+        quantile_preds = self._forward(
+            context=context,
+            group_ids=group_ids,
+            future_covariates=future_covariates,
+            num_output_patches=num_output_patches,
+        )
+
+        # reshape quantile_preds to (batch, time, vars, quantiles)
+        quantile_preds = quantile_preds.view(
+            batch_size,
+            n_variables,
+            num_output_patches,
+            self.num_quantiles,
+            self.chronos_config.output_patch_size,
+        )
+        quantile_preds = quantile_preds.permute(0, 2, 4, 1, 3).reshape(
+            batch_size,
+            num_output_patches * self.chronos_config.output_patch_size,
+            n_variables,
+            self.num_quantiles,
+        )
+
+        # truncate to future_length
+        quantile_preds = quantile_preds[:, :future_length]
+
+        # truncate to only target variables
+        quantile_preds = quantile_preds[:, :, : self.n_targets, :]
+
+        if not self.probabilistic:
+            quantile_preds = quantile_preds[:, :, :, [self.median_idx]]
+
+        return quantile_preds
+
+
+class Chronos2Model(FoundationModel, HuggingFaceModelMixin):
+    _repo_id = "amazon/chronos-2"
+    _repo_commit = "18128c7b4f3fd286f06d6d4efe1d252f1d2a9a7c"
+
+    _allows_finetuning = True
+
+    def __init__(
+        self,
+        input_chunk_length: int,
+        output_chunk_length: int,
+        output_chunk_shift: int = 0,
+        probabilistic: bool = True,
```

Comment:
It would be great to hide/hadle this behavior from/for the user. My idea:

- By default (`likelihood=None, loss_fn=None`) the model will be deterministic. Behind the hood we set `self.probabilistic = False`
- To make the model probabilistic, let the user provide a `likelihood=QuantileRegression(quantiles=user_choice)`. The user can pick any quantiles of the available ones from Chronos2. E.g. `quantiles=[0.1, 0.5, 0.9]`. Like this, users can control which quantiles they would like to see when using `predict_likelihood_parameters=True`. Behind the hood, we will simply extract the corresponding quantiles from the model output in `forward`.

What do you think?